project.go

```diff
@@ -0,0 +1,391 @@
+// Copyright 2016 EF CTX. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/tsuru/gnuflag"
+	"github.com/tsuru/tsuru/cmd"
+	tsuruerrors "github.com/tsuru/tsuru/errors"
+)
+
+type projectCreate struct {
+	fs       *gnuflag.FlagSet
+	name     string
+	platform string
+	team     string
+	plan     string
+	envs     commaSeparatedFlag
+}
+
+func (*projectCreate) Info() *cmd.Info {
+	return &cmd.Info{
+		Name: "project-create",
+		Desc: "creates a remote project in the tranor server",
+	}
+}
+
+func (c *projectCreate) Run(ctx *cmd.Context, client *cmd.Client) error {
+	if c.name == "" || c.platform == "" {
+		return errors.New("please provide the name and the platform")
+	}
+	config, err := loadConfigFile()
+	if err != nil {
+		return errors.New("unable to load environments file, please make sure that tranor is properly configured")
+	}
+	err = c.envs.validate(config.envNames())
+	if err != nil {
+		return fmt.Errorf("failed to load environments: %s", err)
+	}
+	envs := c.getEnvironmentsByName(config.Environments, c.envs.Values())
+	apps, err := c.createApps(envs, client)
+	if err != nil {
+		return err
+	}
+	err = c.setCNames(apps, client)
+	if err != nil {
+		appNames := make([]string, len(apps))
+		for i, app := range apps {
+			appNames[i] = app["name"]
+		}
+		deleteApps(appNames, client)
+		return fmt.Errorf("failed to configure project %q: %s", c.name, err)
+	}
+	fmt.Fprintf(ctx.Stdout, "successfully created the project %q!\n", c.name)
+	if gitRepo := apps[0]["repository_url"]; gitRepo != "" {
+		fmt.Fprintf(ctx.Stdout, "Git repository: %s\n", gitRepo)
+	}
+	return nil
+}
+
+func (c *projectCreate) Flags() *gnuflag.FlagSet {
+	if c.fs == nil {
+		c.fs = gnuflag.NewFlagSet("project-create", gnuflag.ExitOnError)
+		c.fs.StringVar(&c.name, "name", "", "name of the project")
+		c.fs.StringVar(&c.name, "n", "", "name of the project")
+		c.fs.StringVar(&c.platform, "platform", "", "platform of the project")
+		c.fs.StringVar(&c.platform, "l", "", "platform of the project")
+		c.fs.StringVar(&c.team, "team", "", "team that owns the project")
+		c.fs.StringVar(&c.team, "t", "", "team that owns the project")
+		c.fs.StringVar(&c.plan, "plan", "", "plan to use for the project")
+		c.fs.StringVar(&c.plan, "p", "", "plan to use for the project")
+		c.fs.Var(&c.envs, "envs", "comma-separated list of environments to use (defaults to dev,qa,stage,production)")
+		c.fs.Var(&c.envs, "e", "comma-separated list of environments to use (defaults to env,qa,stage,production)")
+		c.envs.Set("dev,qa,stage,production")
```

Comment:
Shouldn't we retrieve this from the config file?
Considering we don't retrieve from the config file,  we should isolate this hard-coded environments somewhere else (constant?). This way if anyone wants to use Tranor for other purposes,
BTW: I strongly suggest we call "production" -> "prod" (everywhere, of course, including the config file - so we keep consistent with an abbreviation for all environments)